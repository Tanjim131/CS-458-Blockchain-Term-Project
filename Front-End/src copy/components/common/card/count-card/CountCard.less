.expiry-range-card {
    box-shadow: 0px 2px 4px #00000029;
    .ant-card-body {
        padding: 0;
        background-color: #FFFFFF;

        .expiry-range-card-title {
            text-align: center;
            font-size: 16px;
            margin: 0;
            padding-top: 16px;
        }
        .expiry-range-card-title,.ant-typography-ellipsis{
            color: #722ED1;
        }
        .expiry-range-card-title-Free {
            color: #FFE58F;
        }
        .expiry-range-card-title-Other {
            color: #646464;
        }
        .expiry-range-card-title-Application {
            color: #0CA789;
        }
        .expiry-range-card-title-Tuition,.expiry-range-card-title-Tuition-university {
            color: #722ED1;
        }
        .expiry-range-card-title-Verification {
            color: #0050B3;
        }

        .expiry-range-card-total-count {
            text-align: center;
            font-size: 24px;
            font-weight: bold;
            margin: 0;
            padding-bottom: 8px;
        }
    }

    .circular-progress{
        text-align: center;
        display: block;
        padding: 16px 0px;
        border-top: 1px solid #D2D2D280;
        border-bottom: 1px solid #D2D2D280;

        .circular-progress-content {
            color: #282828;
            .remaining-time {
                font-size: 24px;
            }
            .time-format {
                font-size: 14px;
            }
            .today-format {
                font-size: 16px;
            }
        }
    }
    .circular-progress-Application {
        color: #F5222D;
    }
    .circular-progress-Tuition {
        color: #722ED1;
    }
    .circular-progress-Verification {
        color: #0050B3;
    }

    .view-detail-text-button {
        display: block;
        color: #4696FF;
        cursor: pointer;
        text-align: center;
        padding: 16px 0px;
    }
}