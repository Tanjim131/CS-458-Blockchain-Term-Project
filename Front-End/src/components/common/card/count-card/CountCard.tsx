import React, { FC } from "react";
import { Card, Progress } from "antd";

import translate from "../../../../../i18n/translate";
import "./CountCard.less";
import Link from "next/link";

interface ICountCard {
    title?: any;
    coinsLeft: number;
    timeLeft: number;
    timeFormat: string;
    ringColor?: string;
    CoinDetailType?: number;
    className?: string;
    CoinDetail?: boolean;
    transactionId?: string;
    walletId: string;
}

const CountCard: FC<ICountCard> = (props: ICountCard) => {
    const { title, coinsLeft, timeLeft, timeFormat, ringColor, CoinDetailType, className, CoinDetail, transactionId,walletId } = props;

    return (
        <Card className={`expiry-range-card ${className}`}>
            <p className={`expiry-range-card-title expiry-range-card-title-${title}`}>
                {title}
            </p>
            <p className="expiry-range-card-total-count">
                {coinsLeft}
            </p>
            <Progress
                type="circle"
                className={"circular-progress circular-progress-" + title}
                percent={100}
                strokeWidth={3}
                strokeColor={timeLeft <= 7 ? "#F5222D" : ringColor}
                width={82}
                format={() => (
                    <div className="circular-progress-content">
                        {timeLeft !== 0
                            ? <>
                                <div className="remaining-time">{timeLeft < 10 ? "0" + timeLeft : timeLeft}</div>
                                <div className="time-format">{timeFormat}</div>
                            </>
                            : <div className="time-format">{translate("today")}</div>}
                    </div>
                )}
            />
        </Card>
    )
}

export default CountCard;
