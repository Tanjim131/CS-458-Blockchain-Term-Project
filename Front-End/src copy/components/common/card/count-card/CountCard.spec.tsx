import React from 'react';
import { shallow } from "enzyme";
import CountCard from "./CountCard";
import { shallowToJson } from "enzyme-to-json";

describe('CountCard Component', () => {
    let wrapper: any;
    const mockFunc = jest.fn();

    beforeAll(() => {
        wrapper = shallow(
            <CountCard
            title = "title"
            coinsLeft = {2}
            timeLeft = {12345}
            timeFormat = "dd/mm/yy"
            ringColor = "color"
            handleViewDetails = {mockFunc}
            />
        )
    })

    it('Check if component exists or not and capture snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    })
})