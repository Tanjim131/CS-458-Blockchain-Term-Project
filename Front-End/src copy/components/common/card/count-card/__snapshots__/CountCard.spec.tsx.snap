// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountCard Component Check if component exists or not and capture snapshot 1`] = `
<Card
  className="expiry-range-card undefined"
>
  <p
    className="expiry-range-card-title expiry-range-card-title-title"
  >
    title
  </p>
  <p
    className="expiry-range-card-total-count"
  >
    2
  </p>
  <Progress
    className="circular-progress circular-progress-title"
    format={[Function]}
    percent={100}
    showInfo={true}
    size="default"
    strokeColor="color"
    strokeLinecap="round"
    strokeWidth={3}
    trailColor={null}
    type="circle"
    width={82}
  />
</Card>
`;
